sympy/polys/monomials.py

```diff
@@ -274,6 +274,54 @@ def monomial_gcd(A, B):
     """
     return tuple([ min(a, b) for a, b in zip(A, B) ])
 
+def monomial_ngcd(monomials):
+    """
+    Computes the greatest common divisor (GCD) of the exponents for each
+    variable in the monomials.
+
+    Parameters
+    ==========
+
+    monomials : iterable of tuples
+        Represents monomials, where each monomial is a tuple of exponents for variables.
+
+    Returns
+    =======
+
+    tuple
+        The greatest common divisor of the exponents for each variable in the
+        monomials.
+
+    Examples
+    ========
+
+    >>> from sympy.polys.monomials import monomial_ngcd
+    >>> monomial_ngcd([(2, 3), (3, 2), (1, 4), (4, 1)])
+    (1, 1)
+
+    """
+    monomials_list = list(monomials)
```

Comment:
Why does this need to be converted to a list? Can a list not be passed in to begin with?